core-default/src/main/java/me/aurium/beetle/defaults/task/MainThreadlessRunner.java

```diff
@@ -0,0 +1,49 @@
+package me.aurium.beetle.defaults.task;
+
+import me.aurium.beetle.api.task.TaskRunner;
+import me.aurium.beetle.api.util.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+public class MainThreadlessRunner implements TaskRunner {
+
+    private final Executor asyncExecutor;
+
+    private final Logger logger = LoggerFactory.getLogger(MainThreadlessRunner.class);
+
+    public MainThreadlessRunner(Executor asyncExecutor) {
+        this.asyncExecutor = asyncExecutor;
+    }
+
+    @Override
+    public void executeSync(Runnable runnable) {
+        throw new NotImplementedException("no main thread");
```

Comment:
Prefer JDK exceptions (e.g. UnsupportedOperationException) to library-defined or application-defined exceptions

Reply:
Didn't know there was an unsupported operation exception. By the way, big commit coming that throws most of this stuff out.

Reply:
maybe not big but it changes stuff